contracts/MarginCalculator.sol

```diff
@@ -143,26 +178,39 @@ contract MarginCalculator {
      * @param _vault theoretical vault that needs to be checked
      * @return marginRequired the minimal amount of collateral needed in a vault, denominated in collateral
      */
-    function _getMarginRequired(
-        MarginVault.Vault memory _vault,
-        bool _hasShort,
-        bool _hasLong
-    ) internal view returns (FPI.FixedPointInt memory) {
-        FPI.FixedPointInt memory shortAmount = _hasShort ? FPI.fromScaledUint(_vault.shortAmounts[0], BASE) : ZERO;
-        FPI.FixedPointInt memory longAmount = _hasLong ? FPI.fromScaledUint(_vault.longAmounts[0], BASE) : ZERO;
-
-        OtokenInterface otoken = _hasShort
-            ? OtokenInterface(_vault.shortOtokens[0])
-            : OtokenInterface(_vault.longOtokens[0]);
-        bool expired = now > otoken.expiryTimestamp();
-        bool isPut = otoken.isPut();
+    function _getMarginRequired(MarginVault.Vault memory _vault, VaultDetails memory _vaultDetails)
+        internal
+        view
+        returns (FPI.FixedPointInt memory)
+    {
+        FPI.FixedPointInt memory shortAmount = _vaultDetails.hasShort
+            ? FPI.fromScaledUint(_vault.shortAmounts[0], BASE)
```

Comment:
why not store _vault.shortAmounts[0] also in vault details? Then you don't need to pass around the vault right? Does that save gas?

Reply:
Yeah I think so, I tried doing that in a diff branch https://github.com/opynfinance/GammaProtocol/pull/345 but broke many tests and couldn't find time to look at that. Maybe if I had time this weekend otherwise we just deploy this as it is for now.